snc.sh

```diff
@@ -24,6 +24,8 @@ sudo rm -fr /var/lib/libvirt/images/crc-podman.qcow2
 CRC_INSTALL_DIR=crc-tmp-install-data
 rm -fr ${CRC_INSTALL_DIR}
 mkdir ${CRC_INSTALL_DIR}
+setfacl -m u:qemu:rx ${HOME}
+setfacl --default -m u:qemu:rx ${PWD}/${CRC_INSTALL_DIR}
```

Comment:
So looks like what we are trying is to fix the downsteam CI with that `setfacl --default -m u:qemu:rx ${PWD}/${CRC_INSTALL_DIR}` and now require both `setfacl` ?

Reply:
The `qemu` user needs `x` permission for all the dirs in `${PWD}/${CRC_INSTALL_DIR}` and it needs `r` permission for the files it tries to read.

`setfacl -m u:qemu:rx ${HOME}` is equivalent to the `chmod 755 ${HOME}` in the microshift PR, but a lot more fine-grained.
`setfacl --default -m u:qemu:rx ${PWD}/${CRC_INSTALL_DIR}` will be useful if new directories are not created with lax enough permissions. Once again, this is similar to what is being tested in microshift here: https://github.com/openshift/microshift/blob/d85cb5043b82d8f5a865a6afc1b0cf082e73e160/scripts/image-builder/configure.sh#L24-L32

My guess is that we have one machine which by default has 0700 permissions for home dirs, and we also get 0750 or 0700 permissions for new directories or new files.

I think we could be even more specific and use `setfacl -m u:qemu:x` for the directories, and `setfacl -m u:qemu:r` for the file, ,but it's not that important.

Reply:
So I tried to replicate what downstream have and then able to reproduce same even with this patch.
```
$ mkdir -p ${HOME}/jenkins/workspace

$ ls -l
total 0
drwxr-x---. 3 prkumar prkumar 23 Mar 15 15:43 jenkins

$ ls -l jenkins/
total 0
drwxr-x---. 3 prkumar prkumar 17 Mar 15 15:50 workspace

$ ls -l jenkins/workspace/
total 4
drwxrwxr-x. 5 prkumar prkumar 4096 Mar 15 15:52 snc

<snc> $ ./snc.sh
[...]
+ sudo virt-install --name=crc-podman --vcpus=2 --ram=2048 --arch=x86_64 --import --graphics=none '--qemu-commandline=-fw_cfg name=opt/com.coreos/config,file=/home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign' --disk=size=31,backing_store=/var/lib/libvirt/images/fedora-coreos-qemu.x86_64.qcow2 --os-variant=fedora-coreos-stable --noautoconsole --quiet
2023-03-15T15:53:03.426355Z qemu-kvm: -fw_cfg name=opt/com.coreos/config,file=/home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign: can't load /home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign: Failed to open file “/home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign”: Permission denied
```

Now check the acls
```
[prkumar@prkumar-test ~]$ getfacl /home/prkumar/
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/
# owner: prkumar
# group: prkumar
user::rwx
user:qemu:r-x
group::---
mask::r-x
other::---

[prkumar@prkumar-test ~]$ getfacl /home/prkumar/jenkins
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/jenkins
# owner: prkumar
# group: prkumar
user::rwx
group::r-x
other::---

[prkumar@prkumar-test ~]$ getfacl /home/prkumar/jenkins/workspace/
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/jenkins/workspace/
# owner: prkumar
# group: prkumar
user::rwx
group::r-x
other::---

[prkumar@prkumar-test ~]$ getfacl /home/prkumar/jenkins/workspace/snc/
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/jenkins/workspace/snc/
# owner: prkumar
# group: prkumar
user::rwx
group::rwx
other::r-x

[prkumar@prkumar-test ~]$ getfacl /home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/
# owner: prkumar
# group: prkumar
user::rwx
group::rwx
other::r-x
default:user::rwx
default:user:qemu:r-x
default:group::rwx
default:mask::rwx
default:other::r-x

[prkumar@prkumar-test ~]$ getfacl /home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign 
getfacl: Removing leading '/' from absolute path names
# file: home/prkumar/jenkins/workspace/snc/crc-tmp-install-data/fcos-config.ign
# owner: prkumar
# group: prkumar
user::rw-
user:qemu:r-x			#effective:r--
group::rwx			#effective:rw-
mask::rw-
other::r--
```

Now if we notice than intermediate directory like `jenkins/workspace` doesn't have respective `acl` and also no `other` permission which again lead to this issue. So to overcome this issue all the intermediate directory also have the `acl` set or the permission for `other`. I think this will again break the CI as soon we have intermediate directory with these kind of permission.

Reply:
I'm confused, do we have intermediate dirs such as these? Do we have plans to introduce such dirs? CI is green so it currently works?

Reply:
yeah, my bad, in the jenkins CI the dir structure is `/home/hudson/jenkins/workspace/eng-crc-snc-arm/<contents_of_snc_repo>`

Reply:
After looking again at this, i think moving the `ign` file to the `/var/images/...` directory solution as in the PR #667 is longer term and it is not tied to the state of the CI machine

to summarise, since we have different dir structure for the openshift CI and internal CI, the solution that works for both is `setfacl -m u:qemu:rx --recursive $HOME` which adds `rx` permission to the `qemu` user for everything under `$HOME`

now if the workspace dir in the CI machine ever changes then we have to update our `snc.sh` script again and it'll be tied to the dir structure of the CI machine

i think #667 is a better solution for this rather then applying lax permissions to whole of $HOME and closely coupling `snc.sh` to where its going to run (there might be other places that assumes certain dir structure in `snc.sh` but i am not looking at those for now)